src/libumf_macosx.c

```diff
@@ -0,0 +1,29 @@
+/*
+ *
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ */
+
+#include "memspace_host_internal.h"
+#include "memspace_internal.h"
+#include "provider_tracking.h"
+
+#include <stddef.h>
+
+umf_memory_tracker_handle_t TRACKER = NULL;
```

Comment:
If we really need a file for macosx I'm starting to think that it would be better to have all this initialization logic in a single file with ifdefs vs having 3 separate files for linux, windows and macosx.

Reply:
It was either a new file or checking for some compile option `MACOSX`, because we don't build the OS provider that the HOST memspace depends on.